### PR TITLE
#165412926 send and show notifications

### PR DESCRIPTION
**Pull request by rwajon**
#### What does this PR do?
Allow users to be able to receive notifications
#### Description of Task to be completed?
Have the following endpoints accessible to an authenticated user:
- `GET /api/v1/notifications` :get all notifications
- `GET /api/v1/notifications/seen` :get all seen notifications
- `GET /api/v1/notifications/unseen` :get all unseen notifications
- `GET /api/v1/notifications/:notificationId` :get one notification
- `DELETE /api/v1/notifications/:notificationId` :delete one notification
- `GET /api/v1/notifications/configuration` :get notification configuration
- `POST /api/v1/notifications/configuration` :create notification configuration
- `PUT /api/v1/notifications/configuration` :update notification configuration
- `PUT /api/v1/notifications/:notificationId` :update notification
- `PUT /api/v1/notifications/seen` :update all notifications status of a logged in user
- `PUT /api/v1/notifications/unseen` :update all notifications status of a logged in user
- `PUT /api/v1/notifications/:notificationId/seen` :update a specific notification status
- `PUT /api/v1/notifications/:notificationId/unseen` :update a specific notification status
To filter notifications, the following parameters can be appended to the URL: `?limit=<number>` or `?offset=<number`
#### How should this be manually tested?
- After cloning the repo, cd into it and type `npm test`
- Using Postman, firstly make sure you signup then login to get the access token, secondly make sure you pass the access token in the header with the following key name: `access-token`, finally send a JSON object to configure notifications with the following structure: `{
   "config": {
	   	"inApp": {
			"articles": {
				"show": true,
				"on": ["publish", "comment", "like"]
			}
		},
		"email": {
			"articles": {
				"show": true,
				"on": ["publish"]
			}
		}
   }
}` to start testing all the provided endpoints
- Using the mockup,  after running the server by typing `npm run start` or `npm run dev`, firstly create an account by visiting `http://localhost:<PORT>/mockups/signup.html`, then login by visiting `http://localhost:<PORT>/mockups/login.html` to get a token, finally visit `http://localhost:<PORT>/mockups/notifications.html` to start testing all the provided endpoints
#### What are the relevant pivotal tracker stories?
[165412926](https://www.pivotaltracker.com/story/show/165412926)
#### Screenshots
image

### CODE REVIEW

File: src/helpers/mailer/sendMail.js [hunk 1]
```diff
@@ -21,9 +21,12 @@ export default async (to, action, data) => {
           <div style="background:#266cef;padding:10px;color:#ffffff;text-align:center;font-size:34px">
           Authors Haven - Ninjas
           </div>
+<<<<<<< HEAD
```
Comment (e-liyai) [inline src/helpers/mailer/sendMail.js]: remove conflict markers

Comment (rwajon) [reply src/helpers/mailer/sendMail.js]: Oh, thank you. I've fixed that

File: templates/html/notifications.html [hunk 2]
```diff
@@ -121,6 +129,17 @@ <h1 class="primary text-white radius-2 uppercase large-padding center-align">
       }
     }
 
+<<<<<<< HEAD
```
Comment (e-liyai) [inline templates/html/notifications.html]: still, remove conflict markers.

Comment (rwajon) [reply templates/html/notifications.html]: I've fixed this, thank you.

**Pull request merged**